gon/core/point.py

```diff
@@ -59,7 +61,32 @@ def __eq__(self, other: 'Point') -> bool:
         >>> Point(0, 0) == Point(1, 0)
         False
         """
-        return (self._x == other._x and self._y == other._y
+        return (self.x == other.x and self.y == other.y
+                if isinstance(other, Point)
+                else NotImplemented)
+
+    def __le__(self, other: 'Point') -> bool:
+        """
+        Checks if the point is less than or equal to the other.
+        Compares points lexicographically, ``x`` coordinates first.
```

Comment:
because other geometries are compound, they consist of points and for them comparison operators have a meaning similar to Python's `set`, while points are more like `tuple`s of coordinates